### PR TITLE
Tracking api

### PR DESCRIPTION
**Pull request by thejacer87**

### CODE REVIEW

File: src/ClientBase.php [hunk 1]
```diff
@@ -112,6 +112,11 @@ public function get($endpoint, array $headers = [], array $options = [])
                 'headers' => $headers,
             ];
 
+            // Enable debug option on development environment.
+            if ($this->config['env'] === self::ENV_DEVELOPMENT) {
+                $options['debug'] = TRUE;
+            }
```
Comment (krystalcode) [inline src/ClientBase.php]: How did this slip in here? It's already done by `buildClient`.

File: src/Tracking.php [hunk 2]
```diff
@@ -4,11 +4,95 @@
...
+     *   The tracking identification type, either pin or dnc. Defaults to pin.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/tracking/trackingsummary.jsf
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getSummary($id, $type = 'pin', array $options = [])
+    {
```
Comment (krystalcode) [inline src/Tracking.php:29]: Might be good to throw an `InvalidArgumentException` if a non-supported type is given.

File: src/Tracking.php [hunk 3]
```diff
@@ -4,11 +4,95 @@
...
+     * @param string $id
+     *   The tracking identification number, either a pin or dnc.
+     * @param string $type
+     *   The tracking identification type, either pin or dnc. Defaults to pin.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/tracking/trackingsummary.jsf
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
```
Comment (krystalcode) [inline src/Tracking.php]: No exception thrown by this method.

File: src/Tracking.php [hunk 4]
```diff
@@ -4,11 +4,95 @@
...
+     * @param string $id
+     *   The tracking identification number, either a pin or dnc.
+     * @param string $type
+     *   The tracking identification type, either pin or dnc. Defaults to pin.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/tracking/trackingdetails.jsf
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
```
Comment (krystalcode) [inline src/Tracking.php:57]: No exception thrown by this method.

File: src/Tracking.php [hunk 5]
```diff
@@ -4,11 +4,95 @@
...
+    /**
+     * Get the image of the signature provided for a specific parcel.
+     *
+     * @param string $pin
+     *   The tracking pin.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
```
Comment (krystalcode) [inline src/Tracking.php:85]: No exception thrown by this method.

File: src/Tracking.php [hunk 6]
```diff
@@ -4,11 +4,95 @@
...
+    /**
+     * Get the image of the delivery confirmation certificate for a parcel.
+     *
+     * @param string $pin
+     *   The tracking pin.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
```
Comment (krystalcode) [inline src/Tracking.php:106]: No exception thrown by this method.

File: src/Tracking.php [hunk 7]
```diff
@@ -4,11 +4,95 @@
...
+
+    /**
+     * Get the image of the signature provided for a specific parcel.
+     *
+     * @param string $pin
+     *   The tracking pin.
+     * @param array $options
+     *   The options to pass along to the Guzzle Client.
+     *
+     * @return \DOMDocument
```
Comment (krystalcode) [inline src/Tracking.php:84]: Does `getFile` return a `DomDocument`? I though it returns the file contents directly.

**Review by krystalcode — CHANGES_REQUESTED** [review]

File: src/Tracking.php (hunk 7 above)
Comment (thejacer87) [reply src/Tracking.php:84]: yes, getFile gets xml files, getArtifact gets raw files

**Review by krystalcode — DISMISSED** [review]

**Pull request merged**